plugins/inputs/filestat/filestat.go

```diff
@@ -85,12 +92,24 @@ func (f *FileStat) Gather(acc telegraf.Accumulator) error {
 			fileInfo, err := os.Stat(fileName)
 			if os.IsNotExist(err) {
 				fields["exists"] = int64(0)
+				acc.AddFields("filestat", fields, tags)
+				if _, ok := f.missingFiles[fileName]; !ok {
+					f.Log.Warnf("File %q not found", fileName)
+					f.missingFiles[fileName] = struct{}{}
+				}
+				continue
+			} else {
+				delete(f.missingFiles, fileName)
 			}
```

Comment:
I think you can simplify it to
```suggestion
				if ! f.missingFiles[fileName] {
					f.Log.Warnf("File %q not found", fileName)
					f.missingFiles[fileName] = true
				}
				continue
			}
			f.missingFiles[fileName] = false
```